rust/src/container.rs

```diff
@@ -418,6 +423,41 @@ pub fn container_encapsulate(args: Vec<String>) -> CxxResult<()> {
             .await
         })
     })?;
+
+    match &opt.compare_with_build {
+        Some(compare_with_build) => handle.block_on(async {
+            let proxy = containers_image_proxy::ImageProxy::new()
+                .await
+                .expect("c/img-proxy");
+            let oi_old = proxy
+                .open_image(compare_with_build)
+                .await
+                .expect("open_image");
+            let (_, manifest_old) = proxy.fetch_manifest(&oi_old).await.expect("fetch_manifest");
+            let oi_now = proxy
+                .open_image(org_args.last().expect("arguments").as_str())
+                .await
+                .expect("open_image");
+            let (_, new_manifest) = proxy.fetch_manifest(&oi_now).await.expect("fetch_manifest");
+            let diff = crate::manifest_diff(&manifest_old, &new_manifest);
+            let layersum = |layers: &Vec<oci_spec::image::Descriptor>| -> u64 {
+                layers.iter().map(|layer| layer.size() as u64).sum()
+            };
+            let new_total = new_manifest.layers().len();
+            let new_total_size = glib::format_size(layersum(new_manifest.layers()));
+            let n_removed = diff.removed.len();
+            let n_added = diff.added.len();
+            let removed_size = layersum(&diff.removed);
+            let removed_size_str = glib::format_size(removed_size);
+            let added_size = layersum(&diff.removed);
+            let added_size_str = glib::format_size(added_size);
+            println!("Total new layers: {new_total}  Size: {new_total_size}");
```

Comment:
Would also make sense to have a handy `fn print(&self)` on `ManifestDiff` I think so we can share this code with the compose path.